### ISSUE TITLE
Better understand the system requirements of the sf package

### ISSUE THREAD

**Issue by maurolepore** [2DegreesInvesting/tiltPlot#69]
To help banks install titPlot we need to better understand the system requirements of the sf package. 

The easiest way is how we did it ourselves, i.e. install Docker then run the [rocker/geospatial](https://github.com/rocker-org/geospatial) image with something like this:

```bash
docker run -d -p 8787:8787 -e PASSWORD=<some password> rocker/geospatial
```

If they don't to it via Docker, then they would need to install each of the dependencies listed in this [Dockerfile](https://github.com/rocker-org/geospatial/blob/master/Dockerfile).

Most likely this will engage the IT folks and that may take time so it's best to start early and learn from the process.

---

Relates to a [comment by Linda on Slack](https://2investinginitiative.slack.com/archives/C054AT1FW2H/p1695738706370789).

**Comment by AnneSchoenauer** [comment]
Hi @maurolepore, 
Good news: I can install the `pak` package and open it without any problems. However, when I type `pak::pak("2DegreesInvesting/tiltPlot")`, it tells me : 

`Using bundled GitHub PAT. Please add your own PAT using gitcred::gitcred_set()`. Error: error in pak subprocess Caused by error: cannot install packages 2DegreesInvesting/tiltPlot: Cannot query GitHub are you offline?"  And yes I am offline ;) 

I did it with PACTA like this that I was downloading the ZIP-folder of the Repos and locally installed for example r2Dii.analysis package. Can I do this as well here? 

Otherwise I am reaching out to the IT!

Best 
Anne

**Comment by AnneSchoenauer** [comment]
One other thing: I just typed now: 
`install.packages("sf")` and it says that it is installed. How do I know if it works?

**Comment by maurolepore** [comment]
Anne, 

The challenging package is sf,so instead of testing it indirectly via tiltPlot for now I would test it directly. 

Please try this:

install.packages("sf")
library(sf)


The second line is important. My experiments showed that sf installs easily but then errors when you use it with library().

If you get any error please share it.

Then try the same with pak.

install.packages("pak")
pak::pak("sf")
library(sf)

**Comment by maurolepore** [comment]
RE you're offline: You must be online. How could install.packages() have worked if you are offline? Packages come from CRAN which is an online resource.

**Comment by AnneSchoenauer** [comment]
With regard to the offline thing - so the Bundesbank saves packages locally and if I press install.packages I install it from this local folder. Apparently they included there the package "sf". I will now run the code that you suggested above

**Comment by maurolepore** [comment]
RE: with PACTA ... I was downloading the ZIP-folder of the Repos and locally installed ...

That should be necessary to solve this problem.

**Comment by AnneSchoenauer** [comment]
RE: library(sf) 

When I press 
`library(sf)` I get a message "linking to GEOS 3.9.3, GDAL 3.5.2, PROJ 8.2.1; sf_use_s2() is TRUE" - so should be fine. 

I also did 
install.packages("pak")
pak::pak("sf")
library(sf)

I don't get any error messages

**Comment by maurolepore** [comment]
RE: Bundesbank saves packages locally

Yeah, that's what I imagined. At some point their IT connects to internet and installs/saves software in their system so the analysts can later use it offline. So the experiments here would test the experience of the IT folks rather than the ultimate users of the system.

RE: Apparently they included there the package "sf".

Great, if they already have "sf" then it should all work.

**Comment by maurolepore** [comment]
Great, tnanks @AnneSchoenauer for being our road tester. 

@lindadelacombaz, [this comment](https://github.com/2DegreesInvesting/tiltPlot/issues/69#issuecomment-1740685753) suggests sf is no problem for @AnneSchoenauer and then likely no problem for the banks -- although the only way to know for sure is to ask them to test it.

Note that after Anne installed sf, running `library(sf)` throws a success message: 

```r
linking to GEOS 3.9.3, GDAL 3.5.2, PROJ 8.2.1; sf_use_s2() is TRUE
```

What I'm not sure is if that works before installing pak. (By the time Anne run `library(sf)` pak was already installed in her system). But anyway it's clear that at most they would have to install pak first and then let pak do its magic behind the hoods. 

In conclusion, I believe we don't need to think of any nuclear option for now and assume it will all just work and then react after we have evidence that they experience any problem.
